### DISCUSSION TITLE
What is an offset or balancing entry in Nacha file?

### DISCUSSION THREAD

**Discussion by OlgaDezh** [moov-io/ach#1112] [Q&A] [answered]
What is an offset or balancing entry in Nacha file?
When integrating with one of payout processors supporting ACH payouts, I've met an unclear block in Nacha file - File Offset Batch (it consists of a header, entry detail, addenda and control record - as if this was a regular payout record). Could you advice what should I put there? What is the offset (also known as balancing entry record) and what should it look like in case there are multiple entry detail records in the batch header record block?
Is the file offset batch is a separate block that keeps as many entry detail records as it was in the Batch header record block? 
![image](https://user-images.githubusercontent.com/116571094/197541525-2b3fa6d6-93c8-4390-918d-51757cbeb76f.png)
@adamdecaf , @nlakritz , guys, you are experts in Nacha format, please, help me to sort out the question.

**Comment by adamdecaf** [comment, marked as answer]
The offset debits your account at the FI. If you're sending a bunch of credits: c1, c2, c3, etc.. then that offset record would debit `c1.Amount` + `c2.Amount` + `c3.Amount`, which equals the amount of money you're sending out from the bank.

**Reply by OlgaDezh** [reply]
@adamdecaf , should I create just one record with accumulated amount of c1, c2, c3? Or should I create as many entry detail records as credit records (one for c1, one for c2 and 1 for c3)?
Could you explain also what is the FI and what is the purpose to debit the FI after credits?

Please, may be share some links explaining the nature of offset/balancing records with me, I couldn't find anything on this subject, any help is highly appreciated.

**Reply by adamdecaf** [reply]
It sounds like your bank wants on entry with the accumulated amounts of c1, c2, c3. 

Your bank follows "double entry accounting" which requires tracking money movement with two entries that offset. A debit and credit to accounts, which track where money moved. 

They use the offset debit to withdraw funds from your account and send them out among the credits. The two sums need to balance out so no funds is lost/created. 

https://www.investopedia.com/terms/d/double-entry.asp#mntl-sc-block_1-0-21

**Reply by OlgaDezh** [reply]
Oh, I see now what you mean.. The issue is that the specification I have does not describe the structure of each offset element (batch, entry details, control record) - these just cannot coincide with origin records at least because identifier cannot be determined for cumulative record in entry details..
Still, thank you.

**Reply by adamdecaf** [reply]
The batch/entry/control records would follow the same specification as the others. You just need to put values in the offset records that make sense.

**Reply by OlgaDezh** [reply]
Thank you
